pkg/client/unversioned/clientcmd/overrides.go

```diff
@@ -171,7 +188,10 @@ func BindAuthInfoFlags(authInfo *clientcmdapi.AuthInfo, flags *pflag.FlagSet, fl
 // BindClusterFlags is a convenience method to bind the specified flags to their associated variables
 func BindClusterFlags(clusterInfo *clientcmdapi.Cluster, flags *pflag.FlagSet, flagNames ClusterOverrideFlags) {
 	flagNames.APIServer.BindStringFlag(flags, &clusterInfo.Server)
+	// TODO: remove --api-version flag on or after Oct. 5, 2016
 	flagNames.APIVersion.BindStringFlag(flags, &clusterInfo.APIVersion)
+	flagNames.PreferredGroupVersions.BindStringSliceFlag(flags, &clusterInfo.PreferredGroupVersions)
+	flags.MarkDeprecated(FlagAPIVersion, fmt.Sprintf("use --%s instead", FlagPreferredGroupVersions))
```

Comment:
This needs to be conditional based on whether the `longName` for the `APIVersion` flag is non-empty.  Otherwise that flag may not be bound.